### ISSUE TITLE
[BUG] no-rolling-update-strategy does not account for the default Strategy being RollingUpdate

### ISSUE THREAD

**Issue by meganwalker-ibm** [stackrox/kube-linter#824]
**Describe the bug**
The `no-rolling-update-strategy` rule is looking for a DeploymentLike object to have a `strategy.type` field matching `RollingUpdate|Rolling`.

However the default value for this field is `RollingUpdate` making specifying it redundant: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy

However the rule is useful for ensuring we don't set a type of `Recreate`.

I think this rule would be better revised to consider a missing value as being a pass.

**To Reproduce**
Enable `no-rolling-update-strategy`.
Lint two deploymentLike objects. One that does not set a strategy. One that sets RolllingUpdate explicitly.
They are equivalent in effect, and so should be treated the same.

**Sample YAML input**
.kube-linter.yaml
```yaml
checks:
  doNotAutoAddDefaults: true
  include:
    - no-rolling-update-strategy
```

sample.yaml
```yaml
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: app1
spec:
  strategy:
    type: Other
---
apiVersion: apps.openshift.io/v1
kind: DeploymentConfig
metadata:
  name: app2
spec:
---
apiVersion: apps.openshift.io/v1
kind: DeploymentConfig
metadata:
  name: app3
spec:
  strategy:
    type: RollingUpdate
```

**Expected behavior**
Only app1 should be identifies as violating this linting rule. app2 and app3 are directly equivalent to each other.

**Additional context**

**Comment by janisz** [comment]
@meganwalker-ibm You're right. Would you like to contribute a patch?
We need to tweek this regexp  https://github.com/stackrox/kube-linter/blob/72736516775d16dc2e546ad0d99ba65c821f7842/pkg/builtinchecks/yamls/no-rolling-update-strategy.yaml#L12 to allow empty value. You can also add your example to tests

**Comment by Eljees** [comment]
This looks fixed on current `main` (`a8f02d3`). Running your sample verbatim:

```console
$ kube-linter lint --include no-rolling-update-strategy --do-not-auto-add-defaults --format json sample.yaml
found 1 lint error
  app1 -> object has Other strategy type but must match regex ^(RollingUpdate|Rolling)$
```

Only `app1` is reported — `app2` (no `strategy` at all) and `app3` (explicit `RollingUpdate`) both pass, which is the behaviour you asked for.

A plain Deployment with no `strategy` key reports nothing at all:

```console
$ kube-linter lint --include no-rolling-update-strategy --do-not-auto-add-defaults --format json plain.yaml
# Reports: null
```

`tests/checks/no-rolling-update-strategy.yml` already carries both shapes — `app3` is a Deployment without a strategy and `app4` is a `DeploymentConfig` with `spec: {}` — and the bats case for this check expects exactly two reports, so the behaviour is pinned.

I cloned shallow, so I can't say which change did it. Worth closing unless you can still reproduce it on a build newer than the one in the report.
